plugin-server/tests/job-queue-manager.test.ts

```diff
@@ -0,0 +1,70 @@
+import { RetryError } from '@posthog/plugin-scaffold'
+
+import { runRetriableFunction } from '../src/utils/retries'
+import { UUID } from '../src/utils/utils'
+import { PromiseManager } from '../src/worker/vm/promise-manager'
+import { JobQueueManager } from './../src/main/job-queues/job-queue-manager'
+import { EnqueuedJob, Hub, JobName } from './../src/types'
+
+jest.mock('../src/utils/retries')
+
+const mockHub: Hub = {
+    instanceId: new UUID('F8B2F832-6639-4596-ABFC-F9664BC88E84'),
+    promiseManager: new PromiseManager({ MAX_PENDING_PROMISES_PER_WORKER: 1 } as any),
+    JOB_QUEUES: 'fs',
+} as Hub
+
+describe('JobQueueManager', () => {
+    let jobQueueManager: JobQueueManager
+    beforeEach(() => {
+        jest.clearAllMocks()
+        jobQueueManager = new JobQueueManager(mockHub)
+    })
+
+    describe('enqueue()', () => {
+        it('calls runRetriableFunction with the correct parameters', async () => {
+            await jobQueueManager.enqueue(JobName.PLUGIN_JOB, { type: 'foo' } as EnqueuedJob)
+            expect(runRetriableFunction).toHaveBeenCalled()
+            const runRetriableFunctionArgs = jest.mocked(runRetriableFunction).mock.calls[0][0]
```

Comment:
Checking calls this way is a code smell.

Do you mean to do something like this:

```
expect(runRetriableFunction).toHaveBeenCalledWith({
  metricPrefix: 'enqueueJob',
  //...
})
```
Note you can use `expect.any()` and equivalent within the object if there's problematic arguments.

Reply:
I have had zero problems and we use both `expect.any()`, `expect.objectContaining()` extensively. toMatchObject much less so though.